docs/developer-guide/aggregate-streaming-data.rst

```diff
@@ -0,0 +1,111 @@
+.. _aggregate-streaming-data-with-ksql:
+
+Aggregate Streaming Data With KSQL
+##################################
+
+KSQL supports several aggregate functions, like COUNT and SUM. You can use
+these to build stateful aggregates on streaming data. For the full list, see
+:ref:`ksql_aggregate_functions`.
+
+Here are some examples that show how to aggregate data from an inbound stream
+of pageview records, named ``pageviews``.
+
+Aggregate Results in a KSQL Table
+*********************************
+
+The result of an aggregate query in KSQL is always a table, because KSQL
+computes the aggregate for each key, and possibly for each window per key, and
+updates these results as it processes new input data.
+
+Assume that you want to count the number of pageviews per region. The following
+query uses the COUNT function to count the pageviews from the time you start the
+query until you terminate it. The query uses the CREATE TABLE AS SELECT statement,
+because the result of the query is a KSQL table.
+
+.. code:: sql
+
+    CREATE TABLE pageviews_per_region AS
+      SELECT regionid,
+             count(*)
+      FROM pageviews
+      GROUP BY regionid;
+
+Aggregate Over Windows
+**********************
+
+KSQL supports aggregation over tumbling, hopping, and session windows.
+
+.. For more information see :ref:`time-and-windows`.
+
+Count Records Over a Tumbling Window
+====================================
+
+The following query is similar to the previous query, but it computes the

```

Comment:
might want to remove `similar to the previous query` since we have no idea whether they've read the previous example and it's kind of superfluous. how about:

```
This query computes the pageview count per region per minute:
```